File: exercises/week04/part02/Address.java
package part02;

public class Address {

  private String streetAddress;
  private String zip;
  private String municipality;

  public Address() {}

  public Address(String street, String zip, String municipality) {
    this.streetAddress = street;
    this.zip = zip;
    this.municipality = municipality;
  }

  public void setStreetAddress(String streetAddress) {
    this.streetAddress = streetAddress;
  }

  public void setZip(String zip) {
    this.zip = zip;
  }

  public void setMunicipality(String municipality) {
    this.municipality = municipality;
  }

  public String getStreetAddress() {
    return this.streetAddress;
  }

  public String getZip() {
    return this.zip;
  }

  public String getMunicipality() {
    return this.municipality;
  }

  public void print() {
    var info = String.format("Street address: %s\nZIP: %s\nMunicipality: %s",
      this.getStreetAddress(),
      this.getZip(),
      this.getMunicipality()
    );
    System.out.println(info);
  }
}

File: exercises/week05/part02/App/address.cpp
#include "address.h"

Address::Address(void) {}
Address::~Address() {}

std::string Address::street(void) const {
    return this->_street;
}

std::string Address::zip(void) const {
    return this->_zip;
}

std::string Address::municipality(void) const {
    return this->_municipality;
}

void Address::set_street(const std::string& street) {
    this->_street = street;
}

void Address::set_zip(const std::string& zip) {
    this->_zip = zip;
}

void Address::set_municipality(const std::string& municipality) {
    this->_municipality = municipality;
}

void Address::set_address(
        const std::string& street,
        const std::string& zip,
        const std::string& municipality
        ) {
    this->_street = street;
    this->_zip = zip;
    this->_municipality = municipality;
}
